### PR TITLE
WIP: document storage 

### PR DESCRIPTION
**Pull request by adamdecaf**
TODO:

- [ ] Sqlite / MySQL storage implementation
- [ ] Encryption/Decryption prior to storage using AES256

### CODE REVIEW

File: migrations/001_documents.up.sql [hunk 1]
```diff
@@ -0,0 +1,9 @@
+create table documents(
+  document_id varchar(40) primary key not null,
```
Comment (adamdecaf) [inline migrations/001_documents.up.sql:2]: @mfdeveloper508 We need to integrate this with the endpoint that saves PDF's/JSON to the database. The calling code will store this `documentID` for lookups later on.

For example, Moov's stack will store (customerID, documentID) to retrieve all documents for a customerID.

Comment (mfdeveloper508) [reply migrations/001_documents.up.sql:2]: can we add metadata  to documents table? 
it seems that we can support lookup about meta data easily.

Comment (adamdecaf) [reply migrations/001_documents.up.sql:2]: What are you thinking that would look like on the table?

Comment (mfdeveloper508) [reply migrations/001_documents.up.sql:2]: I think that we should use json string for this

Comment (adamdecaf) [reply migrations/001_documents.up.sql:2]: Sure, but there may be multiple values. It's not efficient to store that JSON blob as a string. 

For now it's easier to rely on an external system to handle metadata for each file. Let's get them encrypted first.

**Review by InfernoJJ — APPROVED** [review]
Looks good. 

Only critique would be to use `stretchr` for testing since it has `Suite` functionality thats very similar to `check.v1` and we use it every other project. 

More info: https://github.com/stretchr/testify

**Comment by mfdeveloper508** [comment]
thank you for your advises and your carefulness.
I will use stretchr/testify in next work.
let's use pushed test cases

**Pull request merged**